provider/lib/manager.js

```diff
@@ -548,127 +765,232 @@ module.exports = function (logger, triggerDB, redisClient) {
+		//* wrapper function to ensure that the postChanges() call from the @ibm-cloud/cloudant sdk 
+		//* module will not wait forever on connecting to DB and try to get changes. 
+		//* HINT: The wrapper is necessary because it was detected that the postChanges did not 
+		//*       issue resolve() or reject()  in any cases. So it seems to hang forever 
+		//********************************************************************************************
+		var wrappingPostChanges = () => new Promise( (resolve, reject) => {
+			var  error = false; 
+            //*******************************************************************************************
+			//* setup a timeout value that is little bigger then the postChanges() timeout e.g + 1 sec 
+			//* if this timeout reached send the STRING "WRAPPER_TIMEOUT" as err info
+			//*******************************************************************************************
+			setTimeout( function(){ reject( "WRAPPER_TIMEOUT" ) }, wrapperTimeout); 
+			
+			//******************************************************************************
+            //* use longpoll feed to hold the socket to the trigger configDB open as long as 
+            //* possible to reduce HTTP session start/stop cycles . ( max timeout =  1 min)
+            //* It may be that one response contains multiple change recods !!! 
+            //******************************************************************************
+            var viewPath = constants.VIEWS_DESIGN_DOC + '/' + constants.ALL_TRIGGERS ;
+            var postChangesParams = { 
+                db : self.databaseName,
+                feed : 'longpoll',
+                timeout : postChangesTimeout,
+                filter :  "_view",
+                view: viewPath,
+                since: seq ,
+                includeDocs : true 
+                // query_Params : "worker0 | worker1"  --> no more needed since FRA and SYD rebuilt to single worker0  setup
+              }
+            
+            self.triggerDB.postChanges( postChangesParams)
+         	.then( response => {
+				if(!error) {
+					resolve(response);
+				} 		
+		    })
+			.catch ( err  => {
+				reject(err); 
+			})			
+		});
+
+
+        while ( true ) {
+            try {
+                logger.info(method, "Next trigger configDB read sequence starts on : [", seq, "]");
+                
+                let response = await wrappingPostChanges();	
+                //********************************************************************
+                //* get the last_seq value to use in the next setupFollow() query 
+                //* if not part of response, then let it thrown an exception to end in 
+                //* the catch() to ensure that loop continues. 
+                //********************************************************************
+                if ( Object.keys(response).length === 0  ) {
```

Comment:
I think this will continue to reading the length of `response.result.results` even if there are no keys in `response`

Reply:
In case of the promise.resolve(response) the cloudant SDK ensures that a result element is included: See API doc
https://cloud.ibm.com/apidocs/cloudant?code=go#postchanges-databases-response.  That is why the check is not explicitly done, relying on a provided  promise.reject(err)  in case the response.result element cannot be build in the postChanges() call.